ml/Dockerfile

```diff
@@ -0,0 +1,24 @@
+# first stage
+FROM python:3.10 AS builder
+
+WORKDIR /usr/src/app
+
+COPY requirements.txt .
+COPY . .
+
+# install dependencies to the local user directory (eg. /root/.local)
+RUN pip install --user -r requirements.txt
+
+# prod stage
+FROM python:3.10-slim as prod
+WORKDIR /usr/src/app
+
+# copy only the dependencies installation from the 1st stage image
+COPY --from=builder /root/.local /root/.local
```

Comment:
Admittedly I'm not familiar with how things are commonly done for Python in docker, but why copy this from the other image rather than just doing `pip install` in the main image? I also think we don't need a builder image at all for this one, as long as we don't have build-only dependencies to be installed.

Reply:
Ah I got this piece from this guide https://www.docker.com/blog/containerized-python-development-part-1/

Reply:
Not an expert either, but is it possible that there might need to be more tools installed to build the deps, which could be omitted in the final images? So `FROM python` vs `FROM python-slim`.

Reply:
That's the usual reason for the separate build phases, but I don't think it's actually the case here. (And with the complex ML/GPU dependencies, I feel like just a naive copy might cause trouble as well)